Sources/Automaton.swift

```diff
@@ -137,9 +137,9 @@ public final class Automaton<State, Input>
 
 // MARK: Private
 
-private func _compose<A, B, C>(_ g: @escaping (B) -> C, _ f: @escaping (A) -> B) -> (A) -> C
+private func _compose<A, B, C, D>(_ g: @escaping ((B) -> C), _ f: @escaping ((A, D) -> B)) -> ((A, D) -> C)
```

Comment:
Please change the use of type parameters in alphabetical order for readability, i.e.:

```swift
private func _compose<A, B, C, D>(_ g: @escaping ((C) -> D), _ f: @escaping ((A, B) -> C)) -> ((A, B) -> D)
```